src/components/Events/CreateEvent.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+    <form @submit.prevent="onSubmit">
+    <div class="container">
+        <input v-validate="'required|max:80'" v-model="name"
+        name="name" class="event-name" placeholder="Name of Event">
+        <span class="form-errors" v-show="errors.has('name')">{{ errors.first('name') }}</span>
+        <div class="form">
+            <div class="form-element">
+                <input v-validate="'required'"
+                v-model="date" name="date" type="date" size="50">
+                <input v-validate="'required|max:80'" v-model="location"
+                name="location" type="text" size="60" placeholder="Location">
+            </div>
+            <div class="form-errors">
+                <span v-show="errors.has('date')">{{ errors.first('date') }}</span>
+                <span v-show="errors.has('location')">{{ errors.first('location') }}</span>
+            </div>
+            <div class="form-element">
+                <input v-validate="'required'" v-model="startTime"
+                name="start time" type="time" step="300" size="30">
+                <p>to</p>
+                <input v-validate="'required'" v-model="endTime"
+                name="end time" type="time" step="300" size="30">
+            </div>
+            <div class="form-errors">
+                <span v-show="errors.has('start time')">{{ errors.first('start time') }}</span>
+                <span v-show="errors.has('end time')">{{ errors.first('end time') }}</span>
+            </div>
+            <div class="form-element">
+                <textarea v-validate="'required'" v-model="description" rows="10" cols="100"
+                name="description" placeholder="Description"></textarea>
+            </div>
+            <div class="form-errors">
+                <span v-show="errors.has('description')">{{ errors.first('description') }}</span>
+            </div>
+            <div class="form-element">
+                <textarea v-validate="'required'" v-model="whatToBring" rows="7" cols="100"
+                name="what to bring" placeholder="What to Bring"></textarea>
+            </div>
+            <div class="form-errors">
+                <span v-show="errors.has('what to bring')">
+                    {{ errors.first('what to bring') }}
+                </span>
+            </div>
+        </div>
+        <div class="buttons">
+            <input type="submit" class="button" value="Create">
+        </div>
+    </div>
+    </form>
+</template>
+
+<script>
+import Vue from 'vue';
+import VeeValidate from 'vee-validate';
+
+Vue.use(VeeValidate);
+export default {
+  name: 'NewEvent',
+  data() {
+    return {
+      event: {
+        name: '',
+        date: '',
+        startTime: '',
+        endTime: '',
+        description: '',
+        whatToBring: '',
+      },
+    };
+  },
+  methods: {
+    onSubmit() {
```

Comment:
Is this function called automatically by vee-validate? Also, how do you save the event? Like is it pushed to the vuex store?

Reply:
I didn't write the actual code to submit the newly created code yet, but that will be the function that's called when the "Create" button is clicked - I wasn't sure if it should make an API call to put the event into the backend or if it should be pushed to the vuex store, so I just left it for now.